src/DynamicExpresso.Core/Parsing/Parser.cs

```diff
@@ -2111,7 +2111,26 @@ private static bool CheckIfMethodIsApplicableAndPrepareIt(MethodData method, Exp
 			promotedArgs.AddRange(method.Parameters.Skip(promotedArgs.Count).Select<ParameterInfo, Expression>(x =>
 			{
 				if (x.HasDefaultValue)
-					return Expression.Constant(x.DefaultValue, x.ParameterType);
+				{
+					var parameterType = x.ParameterType;
+					if (x.ParameterType.ContainsGenericParameters)
```

Comment:
Can you check whether it shouldn't be `IsGenericParameter`? With `ContainsGenericParameters`, I'm wondering what would happen with a signature like this:

```
public List<T> MethodWithGenericParamAndDefault<T>(T a, List<T> b = default)
{
   return b;
}

var b = MethodWithGenericParamAndDefault(1);
Assert.Null(b);
```

Reply:
Thanks! I've generalized the solution to work with multi level generic types like `List<List<T>>`